src/types/redis_hash.cc

```diff
@@ -386,5 +387,78 @@ rocksdb::Status Hash::Scan(const Slice &user_key, const std::string &cursor, uin
                            std::vector<std::string> *values) {
   return SubKeyScanner::Scan(kRedisHash, user_key, cursor, limit, field_prefix, fields, values);
 }
+rocksdb::Status Hash::RandField(const Slice &user_key, std::vector<FieldValue> *field_values, uint64_t count, bool uniq,
+                                HashFetchType type) {
+  field_values->clear();
+  std::string ns_key;
+  AppendNamespacePrefix(user_key, &ns_key);
+  HashMetadata metadata(false);
+  rocksdb::Status s = GetMetadata(ns_key, &metadata);
+  if (!s.ok()) return s;
+  std::string prefix_key, next_version_prefix_key;
+  InternalKey(ns_key, "", metadata.version, storage_->IsSlotIdEncoded()).Encode(&prefix_key);
+  InternalKey(ns_key, "", metadata.version + 1, storage_->IsSlotIdEncoded()).Encode(&next_version_prefix_key);
 
+  rocksdb::ReadOptions read_options;
+  LatestSnapShot ss(storage_);
+  read_options.snapshot = ss.GetSnapShot();
+  rocksdb::Slice upper_bound(next_version_prefix_key);
+  read_options.iterate_upper_bound = &upper_bound;
+  storage_->SetReadOptions(read_options);
+
+  uint64_t size = metadata.size;
+  std::vector<std::string> fields;
+  std::vector<std::string> values;
+  auto iter = util::UniqueIterator(storage_, read_options);
+  for (iter->Seek(prefix_key); iter->Valid() && iter->key().starts_with(prefix_key); iter->Next()) {
+    InternalKey ikey(iter->key(), storage_->IsSlotIdEncoded());
+    fields.emplace_back(ikey.GetSubKey().ToString());
+    if (type == HashFetchType::kAll) {
+      values.emplace_back(iter->value().ToString());
+    }
+  }
+
+  // Case 1: Negative count, randomly select elements
+  if (!uniq) {
+    field_values->reserve(count);
+    for (uint64_t i = 0; i < count; i++) {
+      uint64_t index = (std::rand() % size);
```

Comment:
It's better to add a rand seed before `rand()` and use `std::mt19937` the same as below.

Reply:
Seems that we have a `src/common/rand.h`, could we use that @PragmaTwice